### PR TITLE
Use eye glyph instead of checkbox for visibility

### PR DESCRIPTION
**Pull request by Broxzier**
As first mentioned in #16072 and later discussed on Discord, the checkbox for visiblity is now replaced by an eye glyph.

I'm not 100% sure if I made the glyph PNG correctly. It would be nice if someone could verify it.

This is how it looks in-game now:
eye could do that

### CODE REVIEW

**Comment by IntelOrca** [comment]
Should it be used for the header of the column as well?

**Comment by Broxzier** [comment]
Yes, that's clearer than the V. I'm thinking of removing the header for that column entirely. The eyes make its function intuitive, especially considering they are there by default.

**Comment by Broxzier** [comment]
I've now updated the column to show the eye symbol too.

Forest Frontiers 2022-04-17 11-16-05

One thing though is that it doesn't show up correctly for fonts that do not have it, like when you set the language to Arabic or Chinese:
Forest Frontiers 2022-04-17 11-17-57
Forest Frontiers 2022-04-17 11-18-20

I'm not sure how to fix that.

**Comment by IntelOrca** [comment]
@Broxzier most strings use `INLINE_SPRITE` for icons in text. The unicode symbols are currently only going to be loaded from the font. @Gymnasiast suggested that all inline sprites become unicode characters are some point though in which case we would need mixed sprite / TTF rendering.

**Comment by Broxzier** [comment]
Alright, thanks for the explanation. I assumed that the checkmark was a sprite, but it is just better supported in fonts. The railway glyph has the same issue as the eye.

**Comment by Gymnasiast** [comment]
> @Gymnasiast suggested that all inline sprites become unicode characters are some point though in which case we would need mixed sprite / TTF rendering.

@IntelOrca We already do this for some characters, like the checkbox and + and - buttons.

@Broxzier I would keep the column header, it can still be helpful (especially the tooltip) and it’s also consistent with the other headers (BH, CH and L).

**Comment by Broxzier** [comment]
@Gymnasiast The header is there, updated to the eye glyph too. Do you know how to make the eye show up for languages that use fonts?

**Comment by Gymnasiast** [comment]
From a cursory glance, it seems you are only missing an entry in `ShouldUseSpriteForCodepoint()`.

**Comment by Broxzier** [comment]
Thanks, that indeed makes it work. I've added other emoji's with entries in the `UnicodeChar` namespace too, only water (🌊️) seems to render as a question mark instead, and the eye glyph has a black background when used in a window title (I guess because of the black border?).

Eye glyph looks good otherwise, but the park name "Forest 🛤🛣👁🌊️❌" looks as follows:

Forest 🛤🛣👁🌊️❌ 2022-04-21 22-49-12

image

**Comment by Gymnasiast** [comment]
Looks like the outline settings of the eye are not quite correct. I can take a look if you can’t find the cause.

**Comment by Gymnasiast** [comment]
> Also, is there a guide somewhere for how these sprites are best made? Paint may not be the most optimal tool for this.

I always used GIMP, but recent versions have a bug with exported indexing PNGs (they started forcibly reindexing it). So now I post-process those images with a PHP script that I quickly hacked up. Apart from the Esperanto-specific letters (which were created by one of the Esperantists) and the bold and small Russian font (created by the company Akella), I created all of them, so there is no guide.

File: distribution/changelog.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - Fix: [#17080] “Remove litter” cheat does not empty litter bins.
 - Fix: [#17099] Object selection thumbnail box is one pixel too tall.
 - Improved: [#16978] Tree placement is more natural during map generation.
+- Improved: [#16992] The checkbox in the visibility column of the Tile Inspector has been replaced by an eye symbol.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:10]: I think the proper preposition is “with”.

```suggestion
- Improved: [#16992] The checkbox in the visibility column of the Tile Inspector has been replaced with an eye symbol.
```

**Comment by Gymnasiast** [comment]
Overall, this seems to work well. But the eye does look quite odd on banners, as it is too big. Perhaps it’s good to include a smaller version that does work on banners.

**Comment by Broxzier** [comment]
@Gymnasiast: I've now added a tiny eye sprite, that looks much better indeed:

image

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**